src/vconsole/vconsole-setup.c

```diff
@@ -54,6 +54,55 @@ static bool is_vconsole(int fd) {
         return ioctl(fd, TIOCLINUX, data) >= 0;
 }
 
+static bool is_allocated(int fd, int idx) {
```

Comment:
This interface is ugly. If `fd` is specified, `idx` is ignored. Since the only caller uses fd>=0, idx param should be dropped.